architecture/builds.adoc

```diff
@@ -0,0 +1,52 @@
+= Builds
+{product-author}
+{product-version}
+:data-uri:
+:icons:
+:experimental:
+:toc: macro
+:toc-title:
+
+toc::[]
+
+== Overview
+A build is a process of transforming input parameters into resulting object.
```

Comment:
And this is what I was thinking when writing 'resulting object'. Only STI & docker produce images and custom build does not.